### PR TITLE
Stage 4 - Async Programming

### PR DESCRIPTION
**Pull request by rominavarela**
#### What does this PR do?
Apply the topics of the 4th stage in the CC Schedule: Async Programming
1. Handle methods with express
2. Replace fake methods with real data using the `fs` and `av` for music metadata

#### Where should the reviewer start?
1. You can see the implementation of express in `main.js`
2. You can see the implementation of the `fs` in the `FileUtils`
3. You can see the implementation of the music metadata in the `TimeUtils`
4. See promise chains in `musicLibrary`

#### Any background context you want to provide?
Learning objectives are:
- Callbacks
- Promises
- Handling the Express web service requests and responses
- Aync/Await if time

I appreciate if someone has an idea on how to make this less complex, I couldn't find how to use `await` which would be nice for the `getTrackMs` method, so less cascading.

#### Screenshots
screen shot 2017-08-09 at 1 30 46 pm
screen shot 2017-08-09 at 1 30 57 pm
screen shot 2017-08-09 at 1 31 09 pm

### CODE REVIEW

**Comment by renemichel** [comment]
@rominavarela You can do this to use async/await with `getTrackMs` . I already tested and it works like that.
```
const getTrackFile = (index) => new Promise((resolve, reject) => {
  listTrackFiles()
    .then( async trackFiles => {
      if(index >= trackFiles.length) {
        reject('Index out of bones');
      } else {
        const trackFile = trackFiles[index];
        const trackPath = FileUtils.getFilePath(trackFile);
        const trackMs = await TimeUtils.getTrackMs(trackPath);
        const track = new Track(trackFile,trackPath, trackMs);
        const decoratedTrack = track.decorate();
        resolve(decoratedTrack);
      }
    });
});
```

**Comment by rominavarela** [comment]
@renemichel can you update the pr with your code? appreciate it

**Comment by codinronan** [comment]
@renemichel don't change all of them: This section is intended to demonstrate several ways to do async programming and Promises are a huge subject in JS.

**Comment by renemichel** [comment]
@codinronan sure, I just changed the one that Romina requested `getTrackMs`.

**Comment by renemichel** [comment]
Async Code pushed

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 node_modules/
 *.log
+.DS_Store
+tracks/
```
Comment (rominavarela) [inline .gitignore:4]: I was going to add this lines to the stage1, which will be the boilerplate... but I think there is no problem with that

**Pull request merged**